testing/sdk_tasks.py

```diff
@@ -151,7 +169,7 @@ def fn():
         return exit_status
 
     # might not be able to connect to the agent on first try so we repeat until we can
-    shakedown.wait_for(lambda: fn(), noisy=True, timeout_seconds=timeout_seconds)
+    fn()
 
     if exit_status != 0:
```

Comment:
Are we guaranteed that the function `fn` captures `exit_status` from the function scope? Should we not assign `exit_status = fn()` here?

Also, since `fn` will retry for non-zero `exit_status`, the conditional statement with the `RuntimeError` below cannot be reached.

Reply:
We are guaranteed to get the exit status, but I would also agree its better to explicitly assign it like that at the end

As for if this can complete, once again, the confusing shakedown API strikes: https://github.com/dcos/shakedown/blob/master/shakedown/dcos/command.py#L72

`shakedown.run_command` return an exist *status* not an exit *code*. Very confusing. Thus, its True when the exit code is zero, I.E. the opposite of an exit code